sdcv update broke mac CI

it looks like the patch is not applying correctly?
caused by #1269

```* Applying /Users/runner/work/koreader-base/koreader-base/thirdparty/sdcv/sdcv.patch . . .
patching file src/sdcv.cpp
Hunk #1 succeeded at 62 with fuzz 2 (offset -4 lines).
missing header for unified diff at line 18 of patch
can't find file to patch at input line 18
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|                    //"./locale"//< for testing
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
3 out of 3 hunks ignored
make[3]: *** [sdcv-download-prefix/src/sdcv-download-stamp/sdcv-download-patch] Error 1
make[2]: *** [CMakeFiles/sdcv-download.dir/all] Error 2
make[1]: *** [all] Error 2
CMake Error at /Users/runner/work/koreader-base/koreader-base/thirdparty/cmake_modules/DownloadProject.cmake:179 (message):
  Build step for sdcv failed: 2
Call Stack (most recent call first):
  CMakeLists.txt:77 (download_project)
```

It also fails locally with a incremental build, but works on a clean checkout.

```
[1/7] Performing update step for 'sdcv-download'
FAILED: sdcv-download-prefix/src/sdcv-download-stamp/sdcv-download-update 
cd /koreader/base/thirdparty/sdcv/build/arm-kindlepw2-linux-gnueabi/sdcv-src && /usr/bin/cmake -P /koreader/base/thirdparty/sdcv/build/arm-kindlepw2-linux-gnueabi/sdcv-download/sdcv-download-prefix/tmp/sdcv-download-gitupdate.cmake
fatal: bad object 4ae420734990ab9f5ccc038262368256b9323f4a
From https://github.com/Dushistov/sdcv
   0c1b821..6c3b136  gh-pages   -> origin/gh-pages
   958ec35..4ae4207  master     -> origin/master
 * [new tag]         v0.5.3     -> v0.5.3
error: Your local changes to the following files would be overwritten by checkout:
	src/sdcv.cpp
Please commit your changes or stash them before you switch branches.
Aborting
CMake Error at /koreader/base/thirdparty/sdcv/build/arm-kindlepw2-linux-gnueabi/sdcv-download/sdcv-download-prefix/tmp/sdcv-download-gitupdate.cmake:201 (message):
  Failed to checkout tag: '4ae420734990ab9f5ccc038262368256b9323f4a'


ninja: build stopped: subcommand failed.
CMake Error at /koreader/base/thirdparty/cmake_modules/DownloadProject.cmake:179 (message):
  Build step for sdcv failed: 1
Call Stack (most recent call first):
  CMakeLists.txt:77 (download_project)
```

--- a/thirdparty/sdcv/sdcv.patch
+++ b/thirdparty/sdcv/sdcv.patch
@@ -1,9 +1,8 @@
 diff --git a/src/sdcv.cpp b/src/sdcv.cpp
-index 0c75eb1..2e40b6d 100644
+index ca222b8..84ec948 100644
 --- a/src/sdcv.cpp
 +++ b/src/sdcv.cpp
-@@ -66,7 +66,10 @@ using StrArr = ResourceWrapper<gchar *, gchar *, free_str_array>;
- static void list_dicts(const std::list<std::string> &dicts_dir_list, bool use_json);
+@@ -63,7 +63,10 @@ static void list_dicts(const std::list<std::string> &dicts_dir_list, bool use_js
  
  int main(int argc, char *argv[])
  try {
@@ -15,7 +14,7 @@ index 0c75eb1..2e40b6d 100644
  #if ENABLE_NLS
      bindtextdomain("sdcv",
                     //"./locale"//< for testing
-@@ -83,12 +86,19 @@ int main(int argc, char *argv[]) try {
+@@ -84,12 +87,19 @@ try {
      gboolean only_data_dir = FALSE;
      gboolean colorize = FALSE;
  
@@ -36,7 +35,7 @@ index 0c75eb1..2e40b6d 100644
            _("for search use only dictionary with this bookname"),
            _("bookname") },
          { "non-interactive", 'n', 0, G_OPTION_ARG_NONE, &non_interactive,
-@@ -101,7 +111,7 @@ int main(int argc, char *argv[]) try {
+@@ -102,7 +112,7 @@ try {
            _("output must be in utf8"), nullptr },
          { "utf8-input", '1', 0, G_OPTION_ARG_NONE, &utf8_input,
            _("input of sdcv in utf8"), nullptr },
@@ -45,7 +44,7 @@ index 0c75eb1..2e40b6d 100644
            _("use this directory as path to stardict data directory"),
            _("path/to/dir") },
          { "only-data-dir", 'x', 0, G_OPTION_ARG_NONE, &only_data_dir,
-@@ -182,7 +192,9 @@ int main(int argc, char *argv[]) try {
+@@ -183,7 +193,9 @@ try {
          // add bookname to list
          gchar **p = get_impl(use_dict_list);
          while (*p) {

